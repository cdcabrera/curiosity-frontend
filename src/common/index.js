import { helpers } from './helpers';
import { dateHelpers } from './dateHelpers';
import { downloadHelpers } from './downloadHelpers';

/**
 * @namespace Helpers
 * @property {module} General
 * @property {module} Dates
 * @property {module} Downloads
 */
export { helpers as default, helpers, dateHelpers, downloadHelpers };
