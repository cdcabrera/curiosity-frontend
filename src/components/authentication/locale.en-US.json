{
  "curiosity-auth": {
    "pending": "Authenticating...",
    "maintenanceCopy": "We are currently undergoing scheduled maintenance and will be back shortly. Thank you for your patience."
  }
}
