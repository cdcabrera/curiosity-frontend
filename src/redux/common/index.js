import { reduxHelpers } from './reduxHelpers';

/**
 * @memberof Redux State
 * @module Helpers
 * @property {module} ReduxHelpers
 */

export { reduxHelpers as default, reduxHelpers };
