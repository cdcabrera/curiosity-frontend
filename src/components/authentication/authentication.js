import React from 'react';
import PropTypes from 'prop-types';
import { BinocularsIcon } from '@patternfly/react-icons';
import { Maintenance } from '@redhat-cloud-services/frontend-components/Maintenance';
import { NotAuthorized } from '@redhat-cloud-services/frontend-components/NotAuthorized';
import { useMount, useUnmount } from 'react-use';
import { connect, reduxActions, reduxSelectors } from '../../redux';
import { rhsmApiTypes } from '../../types';
import { helpers } from '../../common';
import { routerHelpers } from '../router/router';
import MessageView from '../messageView/messageView';
import { translate } from '../i18n/i18n';
import { useHistory } from '../../hooks/useRouter';

/**
 * An authentication pass-through component.
 *
 * @param {object} props
 * @param {Function} props.authorizeUser
 * @param {string} props.appName
 * @param {Node} props.children
 * @param {Function} props.hideGlobalFilter
 * @param {Function} props.initializeChrome
 * @param {Function} props.onNavigation
 * @param {object} props.session
 * @param {Function} props.setAppName
 * @param {Function} props.t
 * @returns {Node}
 */
const Authentication = ({
  appName,
  authorizeUser,
  children,
  initializeChrome,
  hideGlobalFilter,
  onNavigation,
  session,
  setAppName,
  t
}) => {
  const history = useHistory();
  const { subscriptions: authorized } = session.authorized || {};

  let removeListeners = helpers.noop;

  useMount(async () => {
    if (!authorized) {
      await authorizeUser();
    }

    initializeChrome();
    setAppName(appName);
    hideGlobalFilter();

    const appNav = onNavigation(event => {
      const { routeHref } = routerHelpers.getRouteConfig({ id: event.navId });
      history.push(routeHref);
    });

    removeListeners = () => {
      appNav();
    };

    if (
      (session.errorCodes && session.errorCodes.includes(rhsmApiTypes.RHSM_API_RESPONSE_ERROR_DATA_CODE_TYPES.OPTIN)) ||
      session.status === 418
    ) {
      history.push(routerHelpers.getErrorRoute.path);
    }
  });

  useUnmount(() => {
    removeListeners();
  });

  if (helpers.UI_DISABLED) {
    return (
      <MessageView>
        <Maintenance description={t('curiosity-auth.maintenanceCopy', '...')} />
      </MessageView>
    );
  }

  if (authorized) {
    return <React.Fragment>{children}</React.Fragment>;
  }

  if (session.pending) {
    return <MessageView pageTitle="&nbsp;" message={t('curiosity-auth.pending', '...')} icon={BinocularsIcon} />;
  }

  return (
    <MessageView>
      <NotAuthorized serviceName={helpers.UI_DISPLAY_NAME} />
    </MessageView>
  );
};

/**
 * Prop types.
 *
 * @type {{authorizeUser: Function, onNavigation: Function, setAppName: Function, t: Function, children: Node,
 *     appName: string, initializeChrome: Function, session: object, hideGlobalFilter: Function}}
 */
Authentication.propTypes = {
  appName: PropTypes.string,
  authorizeUser: PropTypes.func,
  children: PropTypes.node.isRequired,
  hideGlobalFilter: PropTypes.func,
  initializeChrome: PropTypes.func,
  onNavigation: PropTypes.func,
  setAppName: PropTypes.func,
  session: PropTypes.shape({
    authorized: PropTypes.shape({
      [routerHelpers.appName]: PropTypes.bool
    }),
    errorCodes: PropTypes.arrayOf(PropTypes.string),
    pending: PropTypes.bool,
    status: PropTypes.number
  }),
  t: PropTypes.func
};

/**
 * Default props.
 *
 * @type {{authorizeUser: Function, onNavigation: Function, setAppName: Function, t: translate, appName: string,
 *     initializeChrome: Function, session: object, hideGlobalFilter: Function}}
 */
Authentication.defaultProps = {
  appName: routerHelpers.appName,
  authorizeUser: helpers.noop,
  hideGlobalFilter: helpers.noop,
  initializeChrome: helpers.noop,
  onNavigation: helpers.noop,
  setAppName: helpers.noop,
  session: {
    authorized: {},
    errorCodes: [],
    pending: false,
    status: null
  },
  t: translate
};

/**
 * Apply actions to props.
 *
 * @param {Function} dispatch
 * @returns {object}
 */
const mapDispatchToProps = dispatch => ({
  authorizeUser: () => dispatch(reduxActions.user.authorizeUser()),
  hideGlobalFilter: isHidden => dispatch(reduxActions.platform.hideGlobalFilter(isHidden)),
  initializeChrome: () => dispatch(reduxActions.platform.initializeChrome()),
  onNavigation: callback => dispatch(reduxActions.platform.onNavigation(callback)),
  setAppName: name => dispatch(reduxActions.platform.setAppName(name))
});

/**
 * Create a selector from applied state, props.
 *
 * @type {Function}
 */
const makeMapStateToProps = reduxSelectors.user.makeUserSession();

const ConnectedAuthentication = connect(makeMapStateToProps, mapDispatchToProps)(Authentication);

export { ConnectedAuthentication as default, ConnectedAuthentication, Authentication };
