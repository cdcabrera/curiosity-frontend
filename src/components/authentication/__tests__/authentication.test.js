import React from 'react';
import { helpers } from '../../../common';
import { Authentication } from '../authentication';
import { rhsmConstants } from '../../../services/rhsm/rhsmConstants';

describe('Authentication Component', () => {
  it('should render a basic component', () => {
    const props = {
      useGetAuthorization: () => ({
        error: false,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: undefined
        }
      })
    };

    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('basic');
  });

  it('should render a component error', () => {
    const props = {
      useGetAuthorization: () => ({
        error: true,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: undefined
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('error');
  });

  it('should allow being disabled', () => {
    const props = {
      isDisabled: true,
      useGetAuthorization: () => ({
        error: false,
        pending: false,
        data: {
          authorized: {
            [helpers.UI_NAME]: true
          },
          errorCodes: [],
          errorStatus: undefined
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('disabled');
  });

  it('should return a redirect on 418 error', () => {
    const props = {
      useGetAuthorization: () => ({
        error: true,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: 418
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('418 error');
  });

  it('should return a redirect on a specific 403 error and error code', () => {
    const props = {
      useGetAuthorization: () => ({
        error: true,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [rhsmConstants.RHSM_API_RESPONSE_ERRORS_CODE_TYPES.OPTIN],
          errorStatus: 403
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('403 redirect error');

    const propsUpdated = component.setProps({
      useGetAuthorization: () => ({
        error: true,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: 403
        }
      })
    });

    expect(propsUpdated).toMatchSnapshot('403 error');
  });

  it('should return a message on 401 error', () => {
    const props = {
      useGetAuthorization: () => ({
        error: true,
        pending: false,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: 401
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component.getByText('You do not have access to Subscriptions')).toMatchSnapshot('401 error');
  });

  it('should render a component pending', () => {
    const props = {
      useGetAuthorization: () => ({
        error: false,
        pending: true,
        data: {
          authorized: {},
          errorCodes: [],
          errorStatus: undefined
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component).toMatchSnapshot('pending');
  });

  it('should render a component authorized', () => {
    const props = {
      useGetAuthorization: () => ({
        error: false,
        pending: false,
        data: {
          authorized: {
            [helpers.UI_NAME]: true
          },
          errorCodes: [],
          errorStatus: undefined
        }
      })
    };
    const component = renderComponent(
      <Authentication {...props}>
        <span className="test">lorem</span>
      </Authentication>
    );

    expect(component.find('span')).toMatchSnapshot('authorized');
  });
});
