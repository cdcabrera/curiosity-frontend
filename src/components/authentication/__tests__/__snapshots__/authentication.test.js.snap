// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Component should allow being disabled: disabled 1`] = `
<authentication
  props="{
  "isDisabled": true,
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions"
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-5"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          Subscriptions
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      <div
        class="pf-c-empty-state ins-c-empty-state__maintenance"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 384 512"
            width="1em"
          >
            <path
              d="M360 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24 0 90.965 51.016 167.734 120.842 192C75.016 280.266 24 357.035 24 448c-13.255 0-24 10.745-24 24v16c0 13.255 10.745 24 24 24h336c13.255 0 24-10.745 24-24v-16c0-13.255-10.745-24-24-24 0-90.965-51.016-167.734-120.842-192C308.984 231.734 360 154.965 360 64c13.255 0 24-10.745 24-24V24c0-13.255-10.745-24-24-24zm-75.078 384H99.08c17.059-46.797 52.096-80 92.92-80 40.821 0 75.862 33.196 92.922 80zm.019-256H99.078C91.988 108.548 88 86.748 88 64h208c0 22.805-3.987 44.587-11.059 64z"
            />
          </svg>
          <h4
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-6"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Maintenance in progress
          </h4>
          <div
            class="pf-c-empty-state__body"
          >
            t(curiosity-auth.maintenanceCopy, ...)
          </div>
        </div>
      </div>
       
    </section>
  </section>
</authentication>
`;

exports[`Authentication Component should render a basic component: basic 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          Subscriptions
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      <div
        class="pf-c-empty-state ins-c-not-authorized "
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            />
          </svg>
          <h5
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-2"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            You do not have access to Subscriptions
          </h5>
          <div
            class="pf-c-empty-state__body"
          >
            Contact your organization administrator(s) for more information or visit 
            <a
              href="./settings/my-user-access"
            >
              My User Access
            </a>
              to learn more about your permissions.
          </div>
          <a
            aria-disabled="false"
            class="pf-c-button pf-m-primary"
            data-ouia-component-id="OUIA-Generated-Button-primary-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            href="."
          >
            Go to landing page
          </a>
        </div>
      </div>
       
    </section>
  </section>
</authentication>
`;

exports[`Authentication Component should render a component authorized: authorized 1`] = `
<span
  class="test"
>
  lorem
</span>
`;

exports[`Authentication Component should render a component error: error 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-3"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          Subscriptions
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      <div
        class="pf-c-empty-state ins-c-not-authorized "
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            />
          </svg>
          <h5
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-4"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            You do not have access to Subscriptions
          </h5>
          <div
            class="pf-c-empty-state__body"
          >
            Contact your organization administrator(s) for more information or visit 
            <a
              href="./settings/my-user-access"
            >
              My User Access
            </a>
              to learn more about your permissions.
          </div>
          <a
            aria-disabled="false"
            class="pf-c-button pf-m-primary"
            data-ouia-component-id="OUIA-Generated-Button-primary-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            href="."
          >
            Go to landing page
          </a>
        </div>
      </div>
       
    </section>
  </section>
</authentication>
`;

exports[`Authentication Component should render a component pending: pending 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-21"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
           
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      <div
        class="pf-c-empty-state fadein"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M416 48c0-8.84-7.16-16-16-16h-64c-8.84 0-16 7.16-16 16v48h96V48zM63.91 159.99C61.4 253.84 3.46 274.22 0 404v44c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32V288h32V128H95.84c-17.63 0-31.45 14.37-31.93 31.99zm384.18 0c-.48-17.62-14.3-31.99-31.93-31.99H320v160h32v160c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32v-44c-3.46-129.78-61.4-150.16-63.91-244.01zM176 32h-64c-8.84 0-16 7.16-16 16v48h96V48c0-8.84-7.16-16-16-16zm48 256h64V128h-64v160z"
            />
          </svg>
          <div
            class="pf-c-empty-state__body"
          >
            t(curiosity-auth.pending, ...)
          </div>
        </div>
      </div>
       
    </section>
  </section>
</authentication>
`;

exports[`Authentication Component should return a message on 401 error: 401 error 1`] = `
<h5
  class="pf-c-title pf-m-lg"
  data-ouia-component-id="OUIA-Generated-Title-20"
  data-ouia-component-type="PF4/Title"
  data-ouia-safe="true"
>
  You do not have access to Subscriptions
</h5>
`;

exports[`Authentication Component should return a redirect on 418 error: 418 error 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <article
      class="pf-c-card"
      data-ouia-component-id="OUIA-Generated-Card-1"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe="true"
      data-test="optinView"
      id=""
    >
      <div
        class="pf-l-flex"
      >
        <div
          class="pf-l-flex pf-m-flex-2"
        >
          <div
            class=""
          >
            <div
              class="pf-c-card__title"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-7"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardTitle, {"appName":"Subscriptions"})
              </h1>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardDescription, {"appName":"Subscriptions"})
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-8"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardSeeTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardSeeDescription)
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-9"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardReportTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardReportDescription)
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-10"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardFilterTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardFilterDescription)
            </div>
            <div
              class="pf-c-card__footer"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group pf-m-action"
                >
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      class="pf-c-form__actions"
                    >
                      <button
                        aria-disabled="true"
                        class="pf-c-button pf-m-primary pf-m-disabled"
                        data-ouia-component-id="OUIA-Generated-Button-primary-3"
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe="true"
                        data-test="optinButtonSubmitDisabled"
                        disabled=""
                        type="button"
                      >
                        t(curiosity-optin.buttonIsActive, {"appName":"Subscriptions"})
                      </button>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="pf-l-flex pf-m-flex-1 pf-m-align-self-center"
        >
          <div
            class=""
          >
            <div
              class="pf-c-card__body"
            >
              <article
                class="pf-c-card curiosity-optin-tour"
                data-ouia-component-id="OUIA-Generated-Card-2"
                data-ouia-component-type="PF4/Card"
                data-ouia-safe="true"
                id=""
              >
                <div
                  class="pf-c-card__header"
                >
                  <div
                    class=""
                  >
                    <img
                      alt="t(curiosity-optin.tourTitleImageAlt)"
                      aria-hidden="true"
                      class="pf-c-brand curiosity-optin-image"
                      src="graph4x.png"
                      srcset="graph4x.png 1064w, graph2x.png 600w"
                    />
                  </div>
                </div>
                <div
                  class="pf-c-card__title"
                >
                  <h3
                    class="pf-c-title pf-m-2xl"
                    data-ouia-component-id="OUIA-Generated-Title-11"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    t(curiosity-optin.tourTitle)
                  </h3>
                </div>
                <div
                  class="pf-c-card__body"
                >
                  t(curiosity-optin.tourDescription)
                </div>
                <div
                  class="pf-c-card__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-secondary uxui-curiosity__button-tour"
                    data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-test="optinButtonTour"
                    type="button"
                  >
                    t(curiosity-optin.buttonTour)
                  </button>
                </div>
              </article>
            </div>
          </div>
        </div>
      </div>
    </article>
  </section>
</authentication>
`;

exports[`Authentication Component should return a redirect on a specific 403 error and error code: 403 error 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-17"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          Subscriptions
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      <div
        class="pf-c-empty-state ins-c-not-authorized "
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            />
          </svg>
          <h5
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-18"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            You do not have access to Subscriptions
          </h5>
          <div
            class="pf-c-empty-state__body"
          >
            Contact your organization administrator(s) for more information or visit 
            <a
              href="./settings/my-user-access"
            >
              My User Access
            </a>
              to learn more about your permissions.
          </div>
          <a
            aria-disabled="false"
            class="pf-c-button pf-m-primary"
            data-ouia-component-id="OUIA-Generated-Button-primary-5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            href="."
          >
            Go to landing page
          </a>
        </div>
      </div>
       
    </section>
  </section>
</authentication>
`;

exports[`Authentication Component should return a redirect on a specific 403 error and error code: 403 redirect error 1`] = `
<authentication
  props="{
  "children": [
    {
      "props": {
        "className": "test",
        "children": "lorem"
      },
      "type": "span"
    }
  ],
  "appName": "subscriptions",
  "isDisabled": false
}"
>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <article
      class="pf-c-card"
      data-ouia-component-id="OUIA-Generated-Card-3"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe="true"
      data-test="optinView"
      id=""
    >
      <div
        class="pf-l-flex"
      >
        <div
          class="pf-l-flex pf-m-flex-2"
        >
          <div
            class=""
          >
            <div
              class="pf-c-card__title"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-12"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardTitle, {"appName":"Subscriptions"})
              </h1>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardDescription, {"appName":"Subscriptions"})
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-13"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardSeeTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardSeeDescription)
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-14"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardReportTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardReportDescription)
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-15"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                t(curiosity-optin.cardFilterTitle)
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              t(curiosity-optin.cardFilterDescription)
            </div>
            <div
              class="pf-c-card__footer"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group pf-m-action"
                >
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      class="pf-c-form__actions"
                    >
                      <button
                        aria-disabled="false"
                        class="pf-c-button pf-m-primary"
                        data-ouia-component-id="OUIA-Generated-Button-primary-4"
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe="true"
                        data-test="optinButtonSubmit"
                        type="button"
                      >
                        t(curiosity-optin.buttonActivate, {"appName":"Subscriptions"})
                      </button>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="pf-l-flex pf-m-flex-1 pf-m-align-self-center"
        >
          <div
            class=""
          >
            <div
              class="pf-c-card__body"
            >
              <article
                class="pf-c-card curiosity-optin-tour"
                data-ouia-component-id="OUIA-Generated-Card-4"
                data-ouia-component-type="PF4/Card"
                data-ouia-safe="true"
                id=""
              >
                <div
                  class="pf-c-card__header"
                >
                  <div
                    class=""
                  >
                    <img
                      alt="t(curiosity-optin.tourTitleImageAlt)"
                      aria-hidden="true"
                      class="pf-c-brand curiosity-optin-image"
                      src="graph4x.png"
                      srcset="graph4x.png 1064w, graph2x.png 600w"
                    />
                  </div>
                </div>
                <div
                  class="pf-c-card__title"
                >
                  <h3
                    class="pf-c-title pf-m-2xl"
                    data-ouia-component-id="OUIA-Generated-Title-16"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    t(curiosity-optin.tourTitle)
                  </h3>
                </div>
                <div
                  class="pf-c-card__body"
                >
                  t(curiosity-optin.tourDescription)
                </div>
                <div
                  class="pf-c-card__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-secondary uxui-curiosity__button-tour"
                    data-ouia-component-id="OUIA-Generated-Button-secondary-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-test="optinButtonTour"
                    type="button"
                  >
                    t(curiosity-optin.buttonTour)
                  </button>
                </div>
              </article>
            </div>
          </div>
        </div>
      </div>
    </article>
  </section>
</authentication>
`;
